docker/api/build.py

```diff
@@ -11,7 +12,47 @@
 
 log = logging.getLogger(__name__)
 
-
+class FileUploadBuffer(object):
+    '''
+    Wrap a file object, buffer, and print progress.
+    '''
+    BUFFER_SIZE = 1024*32
+    def __init__(self, file):
+        self.read_so_far = 0
+        self.num_reads = 0
+        self.starttime = datetime.datetime.utcnow()
+        self.file = file
```

Comment:
I like the idea of this abstraction, however I think reporting total reads is a bit weird.

I think it would be good to first get the full size of `file`, and report "send / total" like the docker-cli does.

This class implements both the iterator interface and part of the file interface. Are both necessary?